packages/checkout-storefront/src/sections/Addresses/utils.ts

```diff
@@ -80,6 +80,16 @@ export const isMatchingAddress = (
   return isEqual(omit(address, "id"), omit(addressToMatch, "id"));
 };
 
+export const getMatchingAddressFromList =
```

Comment:
Moved this here because currently this should only happen for guest form, for auth user it's now handled inside of the updated useEffect in AddressListProvider